homeassistant/components/google_travel_time/sensor.py

```diff
@@ -161,23 +96,42 @@ def run_setup(event):
             if mode is None:
                 options[CONF_MODE] = travel_mode
 
-        titled_mode = options.get(CONF_MODE).title()
-        formatted_name = f"{DEFAULT_NAME} - {titled_mode}"
-        name = config.get(CONF_NAME, formatted_name)
-        api_key = config.get(CONF_API_KEY)
-        origin = config.get(CONF_ORIGIN)
-        destination = config.get(CONF_DESTINATION)
+        api_key = config_entry.data.get(CONF_API_KEY)
+        origin = config_entry.data.get(CONF_ORIGIN)
+        destination = config_entry.data.get(CONF_DESTINATION)
+        name = config_entry.data.get(
+            CONF_NAME, f"{DEFAULT_NAME}: {origin} -> {destination}"
+        )
 
         sensor = GoogleTravelTimeSensor(
             hass, name, api_key, origin, destination, options
         )
         hass.data[DATA_KEY].append(sensor)
 
         if sensor.valid_api_connection:
-            add_entities_callback([sensor])
+            async_add_entities([sensor], True)
 
     # Wait until start event is sent to load this component.
-    hass.bus.listen_once(EVENT_HOMEASSISTANT_START, run_setup)
+    await hass.bus.async_listen_once(EVENT_HOMEASSISTANT_START, run_setup)
+
+
+async def async_setup_platform(
+    hass, config, add_entities_callback, discovery_info=None
+):
+    """Set up the Google travel time platform."""
+    await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={"source": SOURCE_IMPORT},
+        data=config,
+    )
+
+    _LOGGER.info(
```

Comment:
not sure this notice is needed. in future release you can use the invalidating helper and it'll provide a notice for users still using YAML

Reply:
you could also check if any entries and short circuit as that means import completed once or user added manually and import is no longer needed

Reply:
See https://github.com/home-assistant/core/pull/43419#discussion_r536812744 as to why I added this. The import logic will already abort if an entry has been previously imported.